depclean-gradle-plugin/src/main/java/se/kth/depclean/analysis/GradleProjectDependencyAnalysis.java

```diff
@@ -0,0 +1,144 @@
+package se.kth.depclean.analysis;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.component.Artifact;
+
+public class GradleProjectDependencyAnalysis {
+
+  /**
+   * Store all the used declared artifacts (ie. used direct dependencies).
+   */
+  private final Set<ResolvedArtifact> usedDeclaredArtifacts;
+
+  /**
+   * Store all the used undeclared artifacts (ie. used transitive dependencies).
+   */
+  private final Set<ResolvedArtifact> usedUndeclaredArtifacts;
+
+  /**
+   * Store all the unused declared artifacts (ie. unused transitive dependencies).
+   */
+  private final Set<ResolvedArtifact> unusedDeclaredArtifacts;
+
+  /**
+   * Ctor.
+   */
+  public GradleProjectDependencyAnalysis(
+          Set<ResolvedArtifact> usedDeclaredArtifacts,
+          Set<ResolvedArtifact> usedUndeclaredArtifacts,
+          Set<ResolvedArtifact> unusedDeclaredArtifacts) {
+    this.usedDeclaredArtifacts = safeCopy(usedDeclaredArtifacts);
+    this.usedUndeclaredArtifacts = safeCopy(usedUndeclaredArtifacts);
+    this.unusedDeclaredArtifacts = safeCopy(unusedDeclaredArtifacts);
+  }
+
+  /**
+   * To prevent unnecessary and unexpected modification in the set.
+   *
+   * @param set required set.
+   * @return An unmodifiable set corresponding to the provided set.
+   */
+  private Set<ResolvedArtifact> safeCopy(Set<ResolvedArtifact> set) {
+    return (set == null) ? Collections.emptySet()
+        : Collections.unmodifiableSet(new LinkedHashSet<ResolvedArtifact>(set));
+  }
+
+//    /**
+//   * Filter out artifacts with scope other than compile from the set of unused declared artifacts.
+//   *
+//   * @return updated project dependency analysis
+//   * @since 1.3
+//   */
+//  public GradleProjectDependencyAnalysis ignoreNonCompile() {
+//    Set<ResolvedArtifact> filteredUnusedDeclared = new HashSet<>(unusedDeclaredArtifacts);
+////    filteredUnusedDeclared.removeIf(resolvedArtifact -> !resolvedArtifact.getScope().equals(Artifact.SCOPE_COMPILE));
```

Comment:
Will it be used?

Reply:
No, it will be of no use as I ignore scopes using the coordinates of the artifacts and not with any Artifact object itself.

Reply:
So, I will remove it.